apps/examples/testcase/ta_tc/systemio/itc/itc_i2c.c

```diff
@@ -63,17 +60,21 @@ void itc_iotbus_i2c_init_stop_p(void)
 void itc_iotbus_i2c_set_frequency_p(void)
 {
 	int ret = IOTBUS_ERROR_NONE;
-	int szFreqMode[] = { IOTBUS_I2C_STD, IOTBUS_I2C_FAST, IOTBUS_I2C_HIGH };
-	int numModes = sizeof(szFreqMode) / sizeof(int);
+	int freq_mode[] = { IOTBUS_I2C_STD, IOTBUS_I2C_FAST, IOTBUS_I2C_HIGH };
+	int num_modes = sizeof(freq_mode) / sizeof(int);
 	int index = 0;
+	bool check = true;
 	iotbus_i2c_context_h h_i2c = iotbus_i2c_init(g_bus);
 	TC_ASSERT_NEQ("iotbus_i2c_init", h_i2c, NULL);
 
-	for (index = 0; index < numModes; index++) {
-		ret = iotbus_i2c_set_frequency(h_i2c, szFreqMode[index]);
-		TC_ASSERT_EQ_CLEANUP("iotbus_i2c_set_frequency", ret, IOTBUS_ERROR_NONE, iotbus_i2c_stop(h_i2c));
+	for (index = 0; index < num_modes; index++) {
+		ret = iotbus_i2c_set_frequency(h_i2c, freq_mode[index]);
+		if (ret != IOTBUS_ERROR_NONE) {
+			check = false;
+			SYSIO_ITC_PRINT("iotbus_i2c_set_frequency failed for frequency : %d for index : %d\n", freq_mode[index], index);
+		}
 	}
-
+	TC_ASSERT_EQ_CLEANUP("itc_iotbus_i2c_set_frequency_p", check, true, iotbus_i2c_stop(h_i2c));
```

Comment:
Your code is
```
if (fail) {
  stop;
  return;
}
stop;
```
So, stop will be called always. To save code size, below is better I think.
```
stop;
if (fail) {
  return
}
```

Reply:
@sunghan-chang 
yes, we agree with you. 
we have used assert macros as above for following coventions:
1. To increase readability and understanding of testcase source code 
2. To maintain uniformity of assert statement throughout testcases

Reply:
But when you expand assertion macro, you will get duplicate codes (i.e. call to `iotbus_i2c_stop`).
So irrespective of whether `itc_iotbus_i2c_set_frequency_p` tc failed or not. `iotbus_i2c_stop` can be called before line no.77.